e2e/machine_migration_helpers.go

```diff
@@ -0,0 +1,291 @@
+	By(fmt.Sprintf("Verify the Machine authority is %s", authority))
+	Eventually(komega.Object(mapiMachine), capiframework.WaitMedium, capiframework.RetryMedium).Should(
+		HaveField("Status.AuthoritativeAPI", Equal(authority)),
+		fmt.Sprintf("Should have found Machine with status.AuthoritativeAPI:%s", authority),
+	)
+}
+
+func verifyMAPIMachineSynchronizedCondition(mapiMachine *mapiv1beta1.Machine, authority mapiv1beta1.MachineAuthority) {
+	By("Verify the MAPI Machine synchronized condition is True")
+	var expectedMessage string
+	switch authority {
+	case mapiv1beta1.MachineAuthorityMachineAPI:
+		expectedMessage = "Successfully synchronized MAPI Machine to CAPI"
+	case mapiv1beta1.MachineAuthorityClusterAPI:
+		expectedMessage = "Successfully synchronized CAPI Machine to MAPI"
+	default:
+		Fail(fmt.Sprintf("unknown authoritativeAPI type: %v", authority))
+	}
+
+	Eventually(komega.Object(mapiMachine), capiframework.WaitMedium, capiframework.RetryMedium).Should(
+		WithTransform(
+			func(m *mapiv1beta1.Machine) []mapiv1beta1.Condition {
+				return m.Status.Conditions
+			},
+			ContainElement(
+				SatisfyAll(
+					HaveField("Type", Equal(SynchronizedCondition)),
+					HaveField("Status", Equal(corev1.ConditionTrue)),
+					HaveField("Reason", Equal("ResourceSynchronized")),
+					HaveField("Message", Equal(expectedMessage)),
+				),
+			),
+		),
+		fmt.Sprintf("Should have found the expected Synchronized condition for MAPI Machine %s with authority: %s", mapiMachine.Name, authority),
+	)
+}
+
+func verifyMAPIMachinePausedCondition(mapiMachine *mapiv1beta1.Machine, authority mapiv1beta1.MachineAuthority) {
+	var conditionMatcher types.GomegaMatcher
+
+	switch authority {
+	case mapiv1beta1.MachineAuthorityMachineAPI:
+		By("Verify the MAPI Machine is Unpaused")
+		conditionMatcher = SatisfyAll(
+			HaveField("Type", Equal(MAPIPausedCondition)),
+			HaveField("Status", Equal(corev1.ConditionFalse)),
+			HaveField("Reason", Equal("AuthoritativeAPIMachineAPI")),
+			HaveField("Message", ContainSubstring("MachineAPI")),
+		)
+	case mapiv1beta1.MachineAuthorityClusterAPI:
+		By("Verify the MAPI Machine is Paused")
+		conditionMatcher = SatisfyAll(
+			HaveField("Type", Equal(MAPIPausedCondition)),
+			HaveField("Status", Equal(corev1.ConditionTrue)),
+			HaveField("Reason", Equal("AuthoritativeAPINotMachineAPI")),
+			HaveField("Message", ContainSubstring("ClusterAPI")),
+		)
+	default:
+		Fail(fmt.Sprintf("unknown authoritativeAPI type: %v", authority))
+	}
+
+	Eventually(komega.Object(mapiMachine), capiframework.WaitMedium, capiframework.RetryMedium).Should(
+		HaveField("Status.Conditions", ContainElement(conditionMatcher)),
+		fmt.Sprintf("Should have found the expected Paused condition for MAPI Machine %s with authority: %s", mapiMachine.Name, authority),
+	)
+}
+
+func verifyCAPIMachinePausedCondition(capiMachine *clusterv1.Machine, authority mapiv1beta1.MachineAuthority) {
```

Comment:
🎉I like this :D

Reply:
I updated based on my understanding, please help to review if it make sense.